builder/cloudstack/config.go

```diff
@@ -120,6 +121,15 @@ func NewConfig(raws ...interface{}) (*Config, error) {
 		c.TemplateDisplayText = c.TemplateName
 	}
 
+	// If we are not given an explicit keypair or ssh_private_key_file, then create
+	// a temporary one, but only if the temporary_keypair has not been provided and
+	// we are not using ssh_password.
```

Comment:
Hmm... Should this comment be:

```
// If we are not given an explicit keypair or ssh_private_key_file, then create
// a temporary keypair name, but only if the temporary_keypair_name has not
// been provided and we are not using ssh_password.
```

Of do I fail to understand what you are trying to do here?